src/main/java/io/gravitee/policy/generatehttpsignature/GenerateHttpSignaturePolicy.java

```diff
@@ -0,0 +1,171 @@
+     */
+    String checkHeaders(HttpHeaders headers, List<String> configuredHeaders) {
+        if (!isEmpty(configuredHeaders)) {
+            if (!headers.containsAllKeys(configuredHeaders)) {
+                final ArrayList<String> missingHeaders = new ArrayList<>(configuredHeaders);
+                missingHeaders.removeAll(headers.keySet());
+
+                return String.format(
+                    "%s those headers are missing %s",
+                    ERROR_MESSAGE,
+                    missingHeaders.stream().collect(Collectors.joining(", ", "[", "]"))
+                );
+            }
+        } else if (!headers.containsKey(HttpHeaders.DATE)) {
+            return String.format("%s 'Date' header is missing", ERROR_MESSAGE);
+        }
+
+        return null;
+    }
+
+    /**
+     * Build signature from policy configuration.
+     * @param context the API execution context
+     * @param configuredHeaders the headers to add to signature
+     * @return the signature
+     */
+    Signature buildSignatureFromConfiguration(ExecutionContext context, Request request, List<String> configuredHeaders) {
+        final String keyId = context.getTemplateEngine().getValue(configuration.getKeyId(), String.class);
+        return new Signature(
+            keyId,
+            null,
+            configuration.getAlgorithm().getAlg(),
+            null,
+            null,
+            configuredHeaders,
+            configuredHeaders.contains("(expires)") ? configuration.getValidityDuration() * 1000 : null,
+            configuredHeaders.contains("(created") ? request.timestamp() : null,
+            null
+        );
+    }
+
+    /**
+     * Build a signer based on the configured signature and the secret from policy configuration.
+     * @param context the API execution context
+     * @param signature the formerly created signature from configuration
+     * @return the signer able to create the signature with provided secret and algorithm
+     */
+    Signer buildSigner(ExecutionContext context, Signature signature) {
+        String secret = context.getTemplateEngine().getValue(configuration.getSecret(), String.class);
+        final Key key = new SecretKeySpec(secret.getBytes(), signature.getAlgorithm().getJvmName());
+        return new Signer(key, signature);
+    }
+
+    /**
+     * Add the HTTP Signature to the request's headers regarding the policy configuration Http Signature Scheme
+     * @param request the request on which add the header
+     * @param signature the signature to use in the header
+     */
+    void setSignatureHeader(Request request, Signature signature) {
+        if (HttpSignatureScheme.SIGNATURE.equals(configuration.getScheme())) {
+            final String substring = signature.toString().substring(10); // Remove the "Signature " part of the signature
+            // https://tools.ietf.org/id/draft-cavage-http-signatures-12.html#rfc.section.4.1
+            request.headers().set("Signature", substring);
```

Comment:
Should we add HttpHeaders.SIGNATURE constant ? And use it in HttpSignaturePolicy too

Reply:
since it's not a standard one we will not add it to HttpHeaders :)